cmd/server/pactasrv/initiative_invitation.go

```diff
@@ -121,3 +133,65 @@ func (s *Server) ListInitiativeInvitations(ctx context.Context, request api.List
 	}
 	return api.ListInitiativeInvitations200JSONResponse(result), nil
 }
+
+func (s *Server) userIsInitiativeManagerOrAdmin(ctx context.Context, iID pacta.InitiativeID) (bool, error) {
+	actorInfo, err := s.getActorInfoOrFail(ctx)
+	if err != nil {
+		return false, err
+	}
+	if actorInfo.IsAdmin || actorInfo.IsSuperAdmin {
+		return true, nil
+	}
+	iurs, err := s.DB.InitiativeUserRelationshipsByInitiative(s.DB.NoTxn(ctx), iID)
+	if err != nil {
+		return false, oapierr.Internal("failed to list initiative user relationships", zap.Error(err))
+	}
+	for _, iur := range iurs {
+		if iur.User.ID == actorInfo.UserID && iur.Manager {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (s *Server) initiativeInvitationDoAuthzAndAuditLog(ctx context.Context, iID pacta.InitiativeID, iiID pacta.InitiativeInvitationID, action pacta.AuditLogAction) error {
+	actorInfo, err := s.getActorInfoOrFail(ctx)
+	if err != nil {
+		return err
+	}
+	iurs, err := s.DB.InitiativeUserRelationshipsByInitiative(s.DB.NoTxn(ctx), iID)
+	if err != nil {
+		return oapierr.Internal("failed to list initiative user relationships", zap.Error(err))
+	}
+	actorIsInitiativeManager := false
+	for _, iur := range iurs {
+		if iur.User.ID == actorInfo.UserID && iur.Manager {
+			actorIsInitiativeManager = true
+			break
+		}
```

Comment:
optional nit: if there's only one `iur` per user, you can exit early if access is granted or denied

```suggestion
		if iur.User.ID == actorInfo.UserID {
			actorIsInitiativeManager = iur.Manager
			break
		}
```

Reply:
good call.